WalletWasabi/WabiSabi/Models/DecompositionAlgs/Decomposition.cs

```diff
@@ -0,0 +1,59 @@
+using NBitcoin;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Models.DecompositionAlgs
+{
+	public class Decomposition : IEnumerable<Coin>
```

Comment:
This class is a wrapper around a list of coins with some useful methods but i am not 100% sure it is really necessary because the list of coins could be handled by the `RestrictedSetDecomposer` directly.

Reply:
I'm not sure, later when multiple decompositions are compared (they represent points in a search space, and sometimes have notions of locality) this class is more meaningful.

Reply:
I will write some tests and see. If it will be used as a List and nothing more I am planning to remove Decomposition

Reply:
Yeah I think it can just be `ImmutableList<Money>`, if we need it to come back later when adding cost functions then we can just wrap around one of those.

I would prefer if it was immutable, not `List`, as it's important to allow composition of different strategies (see notebook for how that's done), and to compare different intermediate results, both of which will be complicated by a mutable list.

BTW NBitcoin provides `MoneyBag` which we can transform this into if that's useful

Reply:
MoneyBag does not fit here as most of its functions are not required, but sorting is missing for example which would be important to have.